File: backend/TravelApp.Api/Models/TravelPlan.cs
namespace TravelApp.Api.Models;

public class TravelPlan
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public User User { get; set; } = null!;

    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }

    public decimal Budget { get; set; }
    public string? Notes { get; set; }

    public string? ShareToken { get; set; }

    public ICollection<Destination> Destinations { get; set; } = new List<Destination>();
    public ICollection<Activity> Activities { get; set; } = new List<Activity>();
    public ICollection<Expense> Expenses { get; set; } = new List<Expense>();
    public ICollection<ChecklistItem> ChecklistItems { get; set; } = new List<ChecklistItem>();
    public ICollection<ShareToken> ShareTokens { get; set; } = new List<ShareToken>();
}

File: frontend/src/models/TravelPlan.js
export function toTravelPlan(data) {
  return {
    id: Number(data.id),
    title: data.title || '',
    description: data.description || '',
    startDate: data.startDate || '',
    endDate: data.endDate || '',
    budget: Number(data.budget || 0),
    notes: data.notes || '',
    totalExpenses: Number(data.totalExpenses || 0),
    shareToken: data.shareToken || null,
    destinations: data.destinations || [],
    activities: data.activities || [],
    expenses: data.expenses || [],
    checklist: data.checklist || [],
  };
}
